### PR TITLE
[#37] Post and publish events and commands to specific type and all superclasses

### PR DESCRIPTION
**Pull request by annaskulimowska**
#37

### CODE REVIEW

**Comment by dkrawiec** [comment]
Looks good, but we need a test for this. AbstractEventBusTest is probably a good place for this.

**Comment by annaskulimowska** [comment]
As I expected Abstract EventBusTest test didn't execute at each build, so it's not work. That's why tests should be be in the right place, not in source code directory

**Comment by annaskulimowska** [comment]
I have ths same result in class MessageTranslatorTest

**Comment by dkrawiec** [comment]
I have just checked it and It is executed with VertXEventBusTest and JsonMessageTranslatorTest

**Comment by annaskulimowska** [comment]
How you run them?

**Comment by dkrawiec** [comment]
test goal for maven from Intellij. Run Tests in "pl.edu.uj.kimage.eventbus" also works. I don't have maven on system's PATH, but running it with mvn test should have same result as with intellij goal. And you?

**Comment by annaskulimowska** [comment]
I had something wrong with the configuration of Maven, now it's ok

File: eventbus/compatibility-kit/src/main/java/pl/edu/uj/kimage/eventbus/AbstractEventBusTest.java [hunk 1]
```diff
@@ -49,17 +62,30 @@ public void shouldRouteOnlyToOneCommandSubscriber() throws Exception {
...
         }
 
         //Then
         assertThat(result).isEqualTo(EXPECTED_RESULT1);
-        assertThat(futureTask2.isDone()!=futureTask1.isDone()).isEqualTo(true);
+        assertThat(futureTask2.isDone() != futureTask1.isDone()).isEqualTo(true);
     }
 
+    @Test
+    public void shouldPostCommandToParents() throws Exception {
```
Comment (dkrawiec) [inline eventbus/compatibility-kit/src/main/java/pl/edu/uj/kimage/eventbus/AbstractEventBusTest.java]: shouldPostCommandToSuperclassSubscribers I would say

**Comment by dkrawiec** [comment]
Missing test for event

File: eventbus/compatibility-kit/src/main/java/pl/edu/uj/kimage/eventbus/AbstractEventBusTest.java (hunk 1 above)
Comment (annaskulimowska) [inline eventbus/compatibility-kit/src/main/java/pl/edu/uj/kimage/eventbus/AbstractEventBusTest.java]: it is the same

**Comment by annaskulimowska** [comment]
shouldPublishEventToParents?

**Comment by dkrawiec** [comment]
shouldPublishEventToSuperclassSubscribers

**Comment by annaskulimowska** [comment]
as you wish ;) but for me it is the same

**Comment by dkrawiec** [comment]
It's not the same. At first I didn't understand what means parent here. If somebody is not familiar with the topic it is more difficult to understand.

**Comment by annaskulimowska** [comment]
So instead continued to argue I changed names - last commit

**Comment by dkrawiec** [comment]
Cool. Now please add test for Events

**Comment by annaskulimowska** [comment]
I don't nderstand, there is one test for commands and one for events

**Comment by dkrawiec** [comment]
Ok, I see. Approved.

**Pull request merged**